Alert icon (#118)
smartdevicelink/sdl_app_library_guides: docs/Displaying a User Interface/Alerts/index.md

```
!@

### Alert Icon

![Generic - Alert](assets/Generic_alertIcon.png)

> An alert can include a custom or static (built-in) image that will be displayed within the alert. Before you add the image to the alert make sure the image is uploaded to the head unit using the @![iOS]SDLFileManager!@@![android,javaSE,javaEE]//TODO!@. If the image is already uploaded, you can set the `alertIcon` property.

Reviewer: ```suggestion
An alert can include a custom or static (built-in) image that will be displayed within the alert. Before you add the image to the alert make sure the image is uploaded to the head unit using the @![iOS]`SDLFileManager`!@@![android,javaSE,javaEE]//TODO!@. If the image is already uploaded, you can set the `alertIcon` property.
```

```
!@

> ### Alert Icon

Reviewer: Move this below "Buttons" instead

> ###### Alert With Non Default Icon

Reviewer: ```suggestion
```

![Generic - Alert](assets/Generic_alertIcon.png)

> An alert can include a custom or static image that will be displayed within the alert. Before you add the image to the alert make sure the image is uploaded to the head unit. If you are certain the image is already uploaded, you can just set `alertIcon` right away.

Reviewer: ```suggestion
An alert can include a custom or static (built-in) image that will be displayed within the alert. Before you add the image to the alert make sure the image is uploaded to the head unit using the @![iOS]SDLFileManager!@@![android,javaSE,javaEE]//TODO!@. If the image is already uploaded, you can set the `alertIcon` property.
```

@![iOS]
##### Objective-C
```objc

> [self.sdlManager.fileManager uploadArtwork:[SDLArtwork artworkWithImage:[UIImage imageNamed:<#ImageName#>] asImageFormat:SDLArtworkImageFormatPNG] completionHandler:^(BOOL success, NSString * _Nonnull artworkName, NSUInteger bytesAvailable, NSError * _Nullable error) {

Reviewer: I think this example might be misleading. If they put this inline with the rest of the example code, it will fail because it's async. Remove the upload code and just include the `alert.alertIcon = [[SDLImage alloc] initWithName:artworkName isTemplate:YES];` code.

    if(error == nil) {
        alert.alertIcon = [[SDLImage alloc] initWithName:artworkName isTemplate:YES];
    }
}];
```
##### Swift
```swift
self.sdlManager.fileManager.upload(artwork: SDLArtwork(image: UIImage(named: <#ImageName#>)!, name: <#ImageName#>, persistent: false, as: <#T##SDLArtworkImageFormat#>)) { (success, artworkName, bytesAvailable, error) in
    guard error == nil else { return }

>     alert.alertIcon = SDLImage(name: artworkName, isTemplate: true)

Reviewer: Same as above
